src/osp/Active/ActiveScene.cpp

```diff
@@ -121,12 +121,14 @@ void ActiveScene::hier_set_parent_child(ActiveEnt parent, ActiveEnt child)
 
 void ActiveScene::hier_destroy(ActiveEnt ent)
 {
-    auto &entHier = m_registry.get<ACompHierarchy>(ent);
+    auto *pEntHier = &m_registry.get<ACompHierarchy>(ent);
 
-    // recurse into children. this can be optimized more but i'm lazy
-    while (entHier.m_childCount)
+    // Destroy children first recursively, if there is any
+    while (pEntHier->m_childCount > 0)

```

Comment:
check for null before dereferencing.

If you're positive that it'll never be null (which I don't think you are, or it wouldn't be "try_get()", then use an assert that's only enabled in debug builds.